### PR TITLE
feat(spdk): support backing image in spdk

### PR DESCRIPTION
**Pull request by ChanYiLin**
ref: https://github.com/longhorn/longhorn/issues/6341

- Add `BackingImageName` in `SpdkInstanceSpec`, so it can create spdk replica with backing image
- Add proxy api, `longhorn-manager` will access spdk server through the engine proxy, but it don't have engine involved.
```
    rpc SPDKBackingImageCreate(SPDKBackingImageCreateRequest) returns (SPDKBackingImageResponse);
    rpc SPDKBackingImageDelete(SPDKBackingImageDeleteRequest) returns (google.protobuf.Empty);
    rpc SPDKBackingImageGet(SPDKBackingImageGetRequest) returns (SPDKBackingImageResponse);
    rpc SPDKBackingImageList(google.protobuf.Empty) returns (SPDKBackingImageListResponse);
    rpc SPDKBackingImageWatch(google.protobuf.Empty) returns (stream google.protobuf.Empty) {}
```
- Add spdk api, so we can manage backing image in spdk server
```
    rpc BackingImageCreate(BackingImageCreateRequest) returns (BackingImage);
    rpc BackingImageDelete(BackingImageDeleteRequest) returns (google.protobuf.Empty);
    rpc BackingImageGet(BackingImageGetRequest) returns (BackingImage);
    rpc BackingImageList(google.protobuf.Empty) returns (BackingImageListResponse);
    rpc BackingImageWatch(google.protobuf.Empty) returns (stream google.protobuf.Empty) {}
    rpc BackingImageExpose(BackingImageGetRequest) returns (BackingImageExposeResponse) {}
    rpc BackingImageUnexpose(BackingImageGetRequest) returns (google.protobuf.Empty) {}
```

### CODE REVIEW

File: protobuf/imrpc/proxy.proto [hunk 1]
```diff
@@ -307,3 +313,47 @@ message EngineMetricsGetProxyResponse {
...
+
+message SPDKBackingImageCreateRequest {
+    string name = 1;
+    string backing_image_uuid = 2;
+	string disk_uuid = 3;
+	uint64 size = 4;
+    string checksum = 5;
+    string from_address = 6;
+	string src_lvs_uuid = 7;
+}
```
Comment (derekbit) [inline protobuf/imrpc/proxy.proto:325]: Fix the indentation.

File: protobuf/imrpc/proxy.proto [hunk 2]
```diff
@@ -307,3 +313,47 @@ message EngineMetricsGetProxyResponse {
...
+    string name = 1;
+    string disk_uuid = 2;
+}
+
+message SPDKBackingImageSpec {
+	string name = 1;
+	string backing_image_uuid = 2;
+    string disk_uuid = 3;
+	uint64 size = 4;
+	string checksum = 5;
```
Comment (derekbit) [inline protobuf/imrpc/proxy.proto:342]: indentation

File: protobuf/imrpc/proxy.proto [hunk 3]
```diff
@@ -307,3 +313,47 @@ message EngineMetricsGetProxyResponse {
...
+	string error_msg = 4;
+}
+
+message SPDKBackingImageResponse {
+	SPDKBackingImageSpec spec = 1;
+	SPDKBackingImageStatus status = 2;
+}
+
+message SPDKBackingImageListResponse {
+    map<string, SPDKBackingImageResponse> backing_images = 1;
```
Comment (derekbit) [inline protobuf/imrpc/proxy.proto:358]: indentation

File: protobuf/imrpc/proxy.proto [hunk 4]
```diff
@@ -45,6 +45,12 @@ service ProxyEngineService {
 	rpc ReplicaRemove(EngineReplicaRemoveRequest) returns (google.protobuf.Empty);
 	rpc ReplicaModeUpdate(EngineReplicaModeUpdateRequest) returns (google.protobuf.Empty);
 
+    rpc SPDKBackingImageCreate(SPDKBackingImageCreateRequest) returns (SPDKBackingImageResponse);
+    rpc SPDKBackingImageDelete(SPDKBackingImageDeleteRequest) returns (google.protobuf.Empty);
+    rpc SPDKBackingImageGet(SPDKBackingImageGetRequest) returns (SPDKBackingImageResponse);
+    rpc SPDKBackingImageList(google.protobuf.Empty) returns (SPDKBackingImageListResponse);
+    rpc SPDKBackingImageWatch(google.protobuf.Empty) returns (stream google.protobuf.Empty) {}
```
Comment (derekbit) [inline protobuf/imrpc/proxy.proto:52]: indentation

File: protobuf/spdkrpc/spdk.proto [hunk 5]
```diff
@@ -426,6 +436,48 @@ message RestoreStatusResponse {
...
+    string name = 1;
+    string lvs_uuid = 2;
+}
+
+message BackingImageListResponse {
+    map<string, BackingImage> backing_images = 1;
+}
+
+message BackingImageExposeResponse {
+    string exposed_snapshot_lvol_address = 1;
```
Comment (derekbit) [inline protobuf/spdkrpc/spdk.proto:478]: indentation

**Review by derekbit — COMMENTED** [review]
After fixing the indentation, we can merge the PR.

File: protobuf/spdkrpc/spdk.proto (hunk 5 above)
Comment (ChanYiLin) [reply protobuf/spdkrpc/spdk.proto:479]: fixed.

File: protobuf/imrpc/proxy.proto (hunk 4 above)
Comment (ChanYiLin) [reply protobuf/imrpc/proxy.proto:52]: fixed.

File: protobuf/imrpc/proxy.proto (hunk 3 above)
Comment (ChanYiLin) [reply protobuf/imrpc/proxy.proto:358]: fixed.

File: protobuf/imrpc/proxy.proto (hunk 2 above)
Comment (ChanYiLin) [reply protobuf/imrpc/proxy.proto]: fixed.

File: protobuf/imrpc/proxy.proto (hunk 1 above)
Comment (ChanYiLin) [reply protobuf/imrpc/proxy.proto]: fixed.

**Review by derekbit — APPROVED** [review]
LGTM

**Pull request merged**